stablehlo/dialect/StablehloOps.cpp

```diff
@@ -1025,6 +1025,63 @@ LogicalResult DotOp::verify() {
   return success();
 }
 
+// Certain enum values have custom printers that include `<` `>`
+// i.e. `<PHILOX>` This method parses the enum value without `<` `>`.
+template <typename EnumAttrKind, typename SymbolizeFn>
+ParseResult parseRawEnum(OpAsmParser& parser, Attribute& enumAttr,
+                         llvm::StringRef errStr, SymbolizeFn symbolizeFn) {
+  llvm::SMLoc loc = parser.getCurrentLocation();
+  llvm::StringRef enumStr;
+  if (failed(parser.parseOptionalKeyword(&enumStr)))
+    return parser.emitError(loc) << "expected " << errStr << " enum";
+  auto enumValOpt = symbolizeFn(enumStr);
+  if (!enumValOpt.has_value())
+    return parser.emitError(loc)
+           << "invalid " << errStr << " enum value " << enumStr;
+  enumAttr = EnumAttrKind::get(parser.getContext(), enumValOpt.value());
+  return success();
+}
+
+// PrecisionConfig - Optional attribute, print the array as raw enums
+//
+// {precision_config = [#stablehlo<precision DEFAULT>,
+//                      #stablehlo<precision DEFAULT>]}
+// ==> ..., precision = [DEFAULT, DEFAULT]
+void printPrecisionConfig(OpAsmPrinter& p, Operation*,
+                          llvm::Optional<mlir::ArrayAttr> attrOpt) {
+  if (!attrOpt.has_value() || !attrOpt.value()) return;
+
+  p << ", precision = [";
+  llvm::interleaveComma(attrOpt.value(), p, [&](Attribute const& attr) {
+    p << stringifyPrecision(attr.cast<PrecisionAttr>().getValue());
+  });
+  p << ']';
+}
+
+ParseResult parsePrecisionConfig(OpAsmParser& parser, mlir::ArrayAttr& attr) {
+  if (failed(parser.parseOptionalComma())) {
+    return success();  // No precision config specified
+  }
+
+  if (failed(parser.parseKeyword("precision")) || failed(parser.parseEqual()))
+    return failure();
+
+  SmallVector<Attribute> attrs;
+  if (failed(
+          parser.parseCommaSeparatedList(AsmParser::Delimiter::Square, [&]() {
+            attrs.emplace_back();
```

Comment:
`attrs.emplace_back()` returns the element that's been emplaced. Given that, can we replace `attrs.back()` below with `attrs.emplace_back()`?